core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -200,4 +200,28 @@ public String location() {
       return location;
     }
   }
+
+  class SetSnapshotRefs implements MetadataUpdate {
```

Comment:
I think this API should focus on updates to individual refs, rather than updating in bulk. What I added to the OpenAPI REST spec was `SetSnapshotRefUpdate`, which is a JSON object like this:

```json
{
  "action": "set-snapshot-ref",
  "ref": "ref name",
  "type": "tag|branch",
  "snapshot-id": 50120357190375
}
```

These can optionally contain `max-ref-age-ms`, `max-snapshot-age-ms`, and `min-snapshots-to-keep`.